packages/core/src/plugins/basic.ts

```diff
@@ -46,7 +46,7 @@ export const pluginBasic = (): RsbuildPlugin => ({
           },
         });
 
-        const usingHMR = !isProd && config.dev.hmr && target === 'web';
+        const usingHMR = isDev && config.dev.hmr && target === 'web';
```

Comment:
Is it okay to do this change?
 I see the snapshots of webpack and svgr are not as same as before

Reply:
These is also one place of `usingHMR` https://github.com/web-infra-dev/rsbuild/blob/a9995acfabc21135143f87e97a4ae66403e7f7cf/packages/plugin-solid/src/index.ts#L37

Reply:
Yes, it is expected, we should only enable HMR when `mode` is `development`.